### ISSUE TITLE
NullInjectorError for IxActiveModal Service

### ISSUE THREAD

**Issue by katjaKoppensteiner** [siemens/ix#1055]
### What happened?

I am trying to use the Modal with Instance variant
like: https://stackblitz.com/edit/gr8rzg-zmtepd?file=src%2Fapp%2Fmodal-by-instance-content.ts

but in this project we use nx workspace, which is a monobased setup with own modules.
it looks like this:
![image](https://github.com/siemens/ix/assets/109066516/0564598b-aa57-437a-805a-b1f0a8dede30)
in my overview.component.html i have the button which is triggering the modalService.open method for displaying the delete-device.component.html

overview.component.ts:
`  async onDeleteClick(device: Device): Promise<void> {
    this.selectedDevice = device;

    console.log('selectedDevice', this.selectedDevice);

    await this.modalService.open({
      content: DeleteDeviceComponent,
      data: device,
      animation: false,
      size: '480',
      backdrop: true,
      centered: true
    });
  }`

delete-device.ts:
`  constructor(readonly activeModal: IxActiveModal) {}

  close(): void {
    this.activeModal.close('My close response');
  }`

delete-device.html:
`<ix-modal-header> Message headline </ix-modal-header>
<ix-modal-content> Message text lorem ipsum: {{ activeModal.data }} </ix-modal-content>
<ix-modal-footer>
  <ix-button outline class="dismiss-modal" (click)="activeModal.dismiss('dismiss')"> Cancel </ix-button>
  <ix-button class="close-modal" (click)="activeModal.close('okay')"> OK </ix-button>
</ix-modal-footer>`

but it's giving me this error in the console:
![image](https://github.com/siemens/ix/assets/109066516/c5fc597c-3150-4a40-a65a-80a29a7f767f)

I assume you are missing the Injectable property for the services:
https://angular.io/guide/dependency-injection
https://stackoverflow.com/questions/48988513/nullinjectorerror-no-provider-for-class

### What type of frontend frameware are you seeing the problem on?

Angular

### Which version of iX do you use?

v2.0.4

### Code to produce this issue.

```shell
see above
```

**Comment by matthiashader** [comment]
Internal ref [IX-883]

**Comment by Aiderlei** [comment]
We encountered the same problem and the only workaround we found was to do this:
```js
// in the template
<ix-button class="modal-close" (click)="modalHack(modal)">
  Invite User
</ix-button>

// in the component class
modalHack(modal: IxActiveModal) {
  console.log('im called')
  modal.close('foobar')
}
```

**Comment by jul-lam** [comment]
I’m unable to replicate the issue when implementing the modal in the following way:

### Basic Usage:

The modal is utilized as outlined in our instance variant documentation, which involves using our modal service and a modal template component.

1. Create a modal template component that instantiates `IxActiveModal`. The `activeModal` object can be used either within your component’s template or directly in the class. 

    _modal-template.component.ts_
    ![image](https://github.com/siemens/ix/assets/151610373/b1a93599-74aa-48b3-a98c-22e861ac778c)


    _modal-template.component.html_
    ![image](https://github.com/siemens/ix/assets/151610373/43e3666a-74fc-4864-98c6-191fb57b5e23)


2. Our modal service is used to instantiate the modal. A modal template component must be used as `content`. 

    ![image](https://github.com/siemens/ix/assets/151610373/cea49172-40a4-45da-adaf-b98c416470a9)


### Advanced Usage:

If you prefer to configure the modal only once instead of adding the configuration each time the modal is instantiated, a wrapper service for our modal service can be created. You can also handle instance events there.

![image](https://github.com/siemens/ix/assets/151610373/70f90818-d6ee-41d3-a825-263873422fd6)

The wrapper service can be utilized in the same manner as our modal service. 

![image](https://github.com/siemens/ix/assets/151610373/f491c830-b918-4a6c-b514-565a4c5617f4)

It’s not advisable to create a custom modal component that combines the service and template functionalities, as this can lead to the errors mentioned above. Please retry with a service wrapper. If this method is insufficient or unsuccessful, feel free to comment.

**Comment by jul-lam** [comment]
I will close the issue. If anything is still not working, you can simply open a new issue with a reference to this one.
